pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+    feed_id: str
+    price: Price
+    ema_price: Price
+    vaa: str
+
+
+
+class HermesClient:
+    def __init__(self, feed_ids: list[str], endpoint=HERMES_ENDPOINT_HTTPS, ws_endpoint=HERMES_ENDPOINT_WSS):
+        self.feed_ids = feed_ids
+        self.pending_feed_ids = feed_ids
+        self.prices_dict: dict[str, PriceFeed] = {}
+        self.client = httpx.AsyncClient()
+        self.endpoint = endpoint
+        self.ws_endpoint = ws_endpoint
+
+    async def get_price_feed_ids(self) -> list[str]:
+        """
+        Queries the Hermes https endpoint for a list of the IDs of all Pyth price feeds.
+        """
+
+        url = os.path.join(self.endpoint, "price_feed_ids")
+        
+        client = httpx.AsyncClient()
+
+        data = (await client.get(url)).json()
+
+        return data
+
+    def add_feed_ids(self, feed_ids: list[str]):
+        self.feed_ids += feed_ids
+        self.feed_ids = list(set(self.feed_ids))
+        self.pending_feed_ids += feed_ids
+
+    @staticmethod
+    def extract_price_feed(data: dict) -> PriceFeed:
+        """
+        Extracts a PriceFeed object from the JSON response from Hermes.
+        """
+        price = Price.from_dict(data["price"])
+        ema_price = Price.from_dict(data["ema_price"])
+        vaa = data["vaa"]
+        price_feed = {
+            "feed_id": data["id"],
+            "price": price,
+            "ema_price": ema_price,
+            "vaa": vaa,
+        }
+        return price_feed
+
+    async def get_pyth_prices_latest(self, feedIds: list[str]) -> list[PriceFeed]:
+        """
+        Queries the Hermes https endpoint for the latest price feeds for a list of Pyth feed IDs.
+        """
+        url = os.path.join(self.endpoint, "latest_price_feeds?")
+        params = {"ids[]": feedIds, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        results = []
+        for res in data:
+            price_feed = self.extract_price_feed(res)
+            results.append(price_feed)
+
+        return results
+
+    async def get_pyth_price_at_time(self, feed_id: str, timestamp: int) -> PriceFeed:
+        """
+        Queries the Hermes https endpoint for the price feed for a Pyth feed ID at a given timestamp.
+        """
+        url = os.path.join(self.endpoint, "get_price_feed")
+        params = {"id": feed_id, "publish_time": timestamp, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        price_feed = self.extract_price_feed(data)
+
+        return price_feed
+
+    async def get_all_prices(self) -> dict[str, PriceFeed]:
+        """
+        Queries the Hermes http endpoint for the latest price feeds for all feed IDs in the class object.
+
+        There are limitations on the number of feed IDs that can be queried at once, so this function queries the feed IDs in batches.
+        """
+        pyth_prices_latest = []
+        i = 0
+        batch_size = 100
```

Comment:
I think you can consider defining this as a class or module-level constant, especially if it's used in multiple places or might change in the future